### ISSUE TITLE
Why do you use nearest method for matching the resolution of (LR, HR) due to CutBlur ?

### ISSUE THREAD

**Issue by star4s** [clovaai/cutblur#19]
I have a question about how to match he resolution of (LR, HR) due to CutBlur.

When I check the code about matching the resolution of (LR, HR) due to CutBlur,
I found using nearest.

# match the resolution of (LR, HR) due to CutBlur
            if HR.size() != LR.size():
                scale = HR.size(2) // LR.size(2)
                LR = F.interpolate(LR, scale_factor=scale, mode="nearest")

Why don't you use bicubic?

Most people use bicubic in super resolution.
Do you have some special things?

I am interest in your CutBlur.
Thank you for your attention.

**Comment by nmhkahn** [comment]
First of all, in our internal analysis, we have found that the nn-based upsample is better than the bicubic (although it's marginal).
Unlike the early-upsample SR methods which mostly upsample via bicubic (e.g. VDSR, SRCNN), our intention of upsampling is not to reconstruct the image in high-res space, rather match the resolution to perform CutBlur.
In this respect, we have decided to provide the raw signal (pixel) itself by not using any high-level interpolation methods.

**Comment by LoveU3tHousand2** [comment]
So,  the LR input , SR output and HR GT have same size ?
it's not for post-upsample SR methods?  eg: LR.shape:120, SR.shape:480, gt.shape:480

**Comment by LoveU3tHousand2** [comment]
If LR and HR have same size at begin , after net.apply SR_size == 4 x LR.size, then how to calculate loss between SR and HR ? downscale HR or what ?  Or just down(CutBlur(up(LR))) ?

**Comment by LoveU3tHousand2** [comment]
Umm, so the method is Cutblur(HR, LR) to replace HR rather LR ?

**Comment by LoveU3tHousand2** [comment]
... Forget it ... I just saw the first layer of these networks ...

**Comment by KaishanLam** [comment]
> 首先，在我们的内部分析中，我们发现基于 nn 的上采样优于双三次（尽管它是边际的）。 与主要通过双三次上采样的早期上采样 SR 方法（例如 VDSR、SRCNN）不同，我们上采样的目的不是在高分辨率空间中重建图像，而是匹配分辨率以执行 CutBlur。 在这方面，我们决定不使用任何高级插值方法来提供原始信号（像素）本身。

After cutblur, LR is now the same size as GT, do you want to downsample it by a factor of 4, f.object (LR, scale_factor=0.25, mode="nearest"), and bring it back to the small size?

**Comment by KaishanLam** [comment]
After cutblur, LR is now the same size as GT, do you want to downsample it by a factor of 4, F.interpolate(LR, scale_factor=0.25, mode="nearest"), and bring it back to the small size?
